gpii/node_modules/flowManager/test/SettingsDataSourceTests.js

```diff
@@ -0,0 +1,280 @@
+/*!
+GPII Settings Data Source
+
+Copyright 2017 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    kettle = require("kettle"),
+    jqUnit = fluid.registerNamespace("jqUnit"),
+    nock = require("nock");
+
+require("flowManager");
+// To reuse gpii.tests.oauth2.invokePromiseProducer()
+fluid.require("%universal/gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/test/js/DataStoreTestsUtils.js");
+
+kettle.loadTestingSupport();
+
+fluid.registerNamespace("gpii.tests.settingsDataSource");
+
+gpii.tests.settingsDataSource.hostname = "http://gpii.net";
+gpii.tests.settingsDataSource.accessTokenUrl = "/access_token";
+gpii.tests.settingsDataSource.untrustedSettingsUrl = "/%userToken/untrusted-settings/%device";
+gpii.tests.settingsDataSource.userToken = "gpii-token-test";
+gpii.tests.settingsDataSource.device = {
+    "OS": "windows 10"
+};
+
+gpii.tests.settingsDataSource.testCases = {
+    nockConfig: {
+        accessToken: {
+            url: gpii.tests.settingsDataSource.accessTokenUrl,
+            data: {
+                "grant_type": "password",
+                "password": "dummy",
+                "client_id": "pilot-computer",
+                "client_secret": "pilot-computer-secret",
+                "username": gpii.tests.settingsDataSource.userToken
+            },
+            status: 200,
+            response: {
+                access_token: "access-token-test-in-settingsDataSource",
+                expiresIn: 3600,
+                token_type: "Bearer"
+            }
+        },
+        untrustedSettings: {
```

Comment:
Probably worth adding the access token Authorization header to the expected request data.